### PR TITLE
Added spawning of all hostile Mobs

### PR DESCRIPTION
**Pull request by geNAZt**
The writing of the attributes for entities was wrong and crashed the client. This commit does fix that. Also the time when the first AddEntity packets are sent has been altered to a point where the player has gotten his/her StartGame Packet to match vanilla behaviour better.

Changed the spawn() Method to return a future to allow async chunkloading to take place when needed. Also i implemented a new dynamic entity implementation detection system. It will automatically scann all entities in the "com.voxelwind.server.game.entities" package and add them to the map when they have the @Spawnable annotation. All entites having the @Spawnable annotation must provide a VoxelwindLevel, Vector3f, Server constructor so the EntitySpawner can construct the entity.

### CODE REVIEW

File: server/src/main/java/com/voxelwind/server/game/level/VoxelwindLevel.java [hunk 1]
```diff
@@ -100,23 +132,36 @@ public int getTime() {
...
         Preconditions.checkNotNull(klass, "klass");
         Preconditions.checkNotNull(position, "position");
-        Preconditions.checkArgument(getBlockIfChunkLoaded(position.toInt()).isPresent(), "entities can not be spawned in unloaded chunks");
-        if (klass.isAssignableFrom(Zombie.class)) {
-            return (T) new ZombieEntity(this, position, server);
-        }
-        throw new IllegalArgumentException("Entity class " + klass.getName() + " not recognized.");
+
+        EntitySpawner entitySpawner;
+        Preconditions.checkNotNull(entitySpawner = ENTITY_SPAWNER.get(klass), "Entity class is not valid");
```
Comment (minecrafter) [inline server/src/main/java/com/voxelwind/server/game/level/VoxelwindLevel.java]: Should really be `Preconditions.checkArgument`

File: server/src/main/java/com/voxelwind/server/network/session/PlayerSession.java [hunk 2]
```diff
@@ -75,6 +78,24 @@
...
+        level.spawn(Ghast.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(MagmaCube.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Silverfish.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Skeleton.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Slime.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Spider.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Witch.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(Zombie.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(ZombiePigman.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
+        level.spawn(ZombieVillager.class, level.getSpawnLocation().add(xDiff += 1.5, 0, 0));
```
Comment (minecrafter) [inline server/src/main/java/com/voxelwind/server/network/session/PlayerSession.java]: Must be removed. Add as a separate debug command instead, or use a plugin.

**Comment by minecrafter** [comment]
Once you resolve conflicts with master, this looks good to me.

**Comment by TheDiamondYT1** [comment]
Or add a test

**Comment by TheDiamondYT1** [comment]
Nvm

**Pull request merged**